src/foam2dolfinx/open_foam_reader.py

```diff
@@ -127,10 +130,17 @@ def create_dolfinx_function(
             the dolfinx function
         """
         self._read_with_pyvista(t=t)
-        self._create_dolfinx_mesh()
-        self.function_space = dolfinx.fem.functionspace(
-            self.dolfinx_mesh, self.mesh_element
-        )
+
+        # Create dolfinx mesh if it doesn't exist
+        if not hasattr(self, "dolfinx_mesh"):
+            self._create_dolfinx_mesh()
+
+        if name == "U":
+            element = self.mesh_vector_element
+        else:
+            element = self.mesh_scalar_element
```

Comment:
Again could add small comment, in the case of reading for "U" a velocity field, a vector element is used etc